builtin/providers/aws/data_source_aws_elastic_beanstalk_solution_stack.go

```diff
@@ -0,0 +1,109 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/service/elasticbeanstalk"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsElasticBeanstalkSolutionStack() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsElasticBeanstalkSolutionStackRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateSolutionStackNameRegex,
+			},
+			"most_recent": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+			// Computed values.
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// dataSourceAwsElasticBeanstalkSolutionStackRead performs the API lookup.
+func dataSourceAwsElasticBeanstalkSolutionStackRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticbeanstalkconn
+
+	nameRegex, nameRegexOk := d.GetOk("name_regex")
```

Comment:
We can do this a different way. If we mark the schema param as required, then we won't need to use `d.GetOk` and it will validate at plan time